### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by munjalSaksham71**
Features: 
1. Profile Card(Static)
2. Created Navlink Card 
3. Post Data and View data
4. Private Routes
[NOTE: For Less Than 1000px the sidebar cards are gone - In follow up PR - I'll Create Hamburger menu for it.]

### CODE REVIEW

File: src/Components/Card/Card.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.card {
+    border-radius: 10px;
+    background-color: #fff;
```
Comment (t007rushi) [inline src/Components/Card/Card.css]: use color variables

File: src/Components/Header/Header.css [hunk 2]
```diff
@@ -8,4 +8,14 @@
     font-weight: 600;
     font-size: x-large;
     color: #000;
```
Comment (t007rushi) [inline src/Components/Header/Header.css]: same here

File: src/Components/SideBar/SideBar.jsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import Card from '../Card/Card';
+import './SideBar.css';
+import { AiOutlineHome, BsPeople, AiOutlineLike, CgProfile, RiUserFollowLine, BsBookmarks } from '../Icon'
+import { NavLink } from 'react-router-dom';
+
+const links = [
```
Comment (t007rushi) [inline src/Components/SideBar/SideBar.jsx]: can put outside

File: src/Screens/HomeScreen/HomeScreen.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  useEffect(() => {
+    (async() => {
+      const data = await getPosts();
+      await postDispatch({type: 'GET_POSTS', payload: data.posts})
+     })()
+  },[])
+
+  const { postState, postDispatch } = usePosts();
+
+  console.log(postState);
```
Comment (t007rushi) [inline src/Screens/HomeScreen/HomeScreen.jsx]: remove console

File: src/reducer/postsReducer.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+export const postsReducer = (state, action) => {
+    switch(action.type){
+        case 'GET_POSTS':
+            return {...state, posts: action.payload};
+        case 'CREATE_POST': 
+            return {...state, posts: action.payload};
+        case 'DELETE_POST':
+            return {...state, posts: action.payload};
```
Comment (t007rushi) [inline src/reducer/postsReducer.js]: can create a single update as all have same code

**Review by t007rushi — APPROVED** [review]
Looks Nice Good folder structure
Look for suggestions

**Review by JantuDeb — APPROVED** [review]
Overall the code and folder structure is looking good.

**Pull request merged**